detection_rules/rule_validators.py

```diff
@@ -713,28 +722,348 @@ def validate_rule_type_configurations(self, data: EQLRuleData, meta: RuleMeta) -
 class ESQLValidator(QueryValidator):
     """Validate specific fields for ESQL query event types."""
 
+    def __init__(self, query: str) -> None:
+        """Initialize the ESQLValidator with the given query."""
+        super().__init__(query)
+        self.esql_unique_fields: list[dict[str, str]] = []
+        self.related_integrations: list[dict[str, str]] = []
+        self.stack_version: str = ""
```

Comment:
Given this is a dataclass and precedence of the other validators, it seems unconventional to `__init__`.